/jds/bin/diff -uprN gnome-panel-2.26.3.old/gnome-panel/panel-run-dialog.c gnome-panel-2.26.3/gnome-panel/panel-run-dialog.c
--- gnome-panel-2.26.3.old/gnome-panel/panel-run-dialog.c	2009-04-19 18:45:09.000000000 +0100
+++ gnome-panel-2.26.3/gnome-panel/panel-run-dialog.c	2009-07-01 18:48:11.982368000 +0100
@@ -1426,6 +1426,12 @@ entry_event (GtkEditable    *entry,
 	char             *temp;
 	int               pos, tmp;
 
+	/* commit the string when IM is enabled */
+	if (GTK_ENTRY (entry)->editable && event->type == GDK_KEY_PRESS && event->length > 0) {
+		if (gtk_im_context_filter_keypress (GTK_ENTRY (entry)->im_context, event))
+			return TRUE;
+	}
+
 	if (event->type != GDK_KEY_PRESS)
 		return FALSE;
 
